File: BACKEND/facturador_backend/src/main/java/com/bap/api/services/api/ApiConfiguracionService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.bap.api.services.api;

import com.bap.api.model.api.ApiConfiguracion;
import com.bap.api.configurate.AbstractEntity;
import com.bap.api.dto.Respuesta;

/**
 *
 * @author ruth
 */
public interface ApiConfiguracionService extends AbstractEntity<ApiConfiguracion> {

    Respuesta solicitudCUFD(String login);

    Respuesta sincronizarFechaHora(String login);

    ApiConfiguracion getConfiguracionVigenteBySucursal(Long idSucursal);

    ApiConfiguracion getConfiguracionVigenteByPuntoVenta(Long idPuntoVenta);

}

File: FRONTEND/facturador-frontend/src/services/ApiConfiguracionService.js
import Api from '@/services/Api'

export default {
    ///ApiConfiguracion
    getApiConfiguracionPorIdEmpresa(idEmpresa){
        return Api().get(`/api/apiConfiguracion/getPorIdEmpresa/${idEmpresa}`);
    },
    postApiConfiguracion(dato){
        return Api().post("/api/apiConfiguracion", dato);
    },
    getConfiguracionVigenteBySucursal(idSucursal){
        return Api().get(`/api/apiConfiguracion/getConfiguracionVigenteBySucursal/${idSucursal}`);
    },
    getConfiguracionVigenteByPuntoVenta(idPuntoVenta){
        return Api().get(`/api/apiConfiguracion/getConfiguracionVigenteByPuntoVenta/${idPuntoVenta}`);
    },
    solicitudCufd(login){
        return Api().get(`/api/apiConfiguracion/solicitudCufd/${login}`);
    },
    sincronizarFechaHora(login){
        return Api().get(`/api/apiConfiguracion/sincronizarFechaHora/${login}`);
    },
}    